### PR TITLE
About user window

### PR DESCRIPTION
**Pull request by agilob**
- [x] #1636 | Contact notes
- [ ] ~~#1189 | Lookup previous aliases~~ can't do now due to ugly DB schema
- [x] ~~#2391 | Profile information~~ - duplicate of public id, avatar and notes
- [x] #2236 | Show friend public key
- [x] make delete button work
- [ ] ~~list all past nicks~~ can't do now due to ugly DB schema
- [x] make autoaccept checkbox work - and dir selector
- [x] WHY ARE MY EYES BLEEDING

First working prototype:

twitchinstallarch15

### CODE REVIEW

**Comment by Zer0-One** [comment]
Can you add some way to get a better look at the user's avatar? qTox displays them so tiny that you can't actually make anything out of the image most of the time.

**Comment by ProMcTagonist** [comment]
@Zer0-One 
+1, avatar should be displayed large in this window.

**Comment by agilob** [comment]
Yes, see the last point.

**Comment by ovalseven8** [comment]
Related issue:
https://github.com/tux3/qTox/issues/2187 | Avatars too small

**Comment by agilob** [comment]
No, that's not related. I purposely used that small avatar because it was more handy at the time.

**Comment by ovalseven8** [comment]
@agilob: Well, I thought that avatars are displayed more large in the new window compared to the current display of the avatars.

**Comment by agilob** [comment]
No, there are two objects representing avatar in qtox, one on friends list - which is scaled down to be smaller and fit on the list - and I used that object in about window, and second avatar which is bigger and you can see it in a chat ;) 

I'll use the bigger one later or load it from disk directly. It's overview/prototype of things I would like to see in the about window, so it's ugly and incomplete. Feel free to suggest more features ;) (buttons are placebo now, they don't work).

**Comment by ghost** [comment]
Options order should be, IMO:
1- path to save files
2- auto accept files checkbox
3- remove history (eventually, see below)

The path selector should be concordant with the one used in the settings:


The history of aliases could become too large to handle as a single-row list. It may be also useful to add more details, like when a particular alias was set and if there was a custom alias hiding it at that time. That, plus the `Remove history` button could go to a separate `History` tab, and the list of aliases could become a scrollable list. More history options (if there's a need for them) could be added to that tab. Like the date when the user was added, last time seen online (that could also go in the main tab), etc.

**Comment by agilob** [comment]
> the list of aliases could become a scrollable list.

I don't get it.

Improved it a bit:
screenshot3

**Comment by agilob** [comment]
I think `WHY ARE MY EYES BLEEDING` can be marked as done.
screenshot4

File: src/widget/about/aboutuser.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    this->setWindowTitle(f->getDisplayedName());
+    ui->userName->setText(f->getDisplayedName());
+    ui->publicKey->setText(QString(f->getToxId().toString()));
+    ui->note->setPlainText(Settings::getInstance().getContactNote(f->getToxId()));
+    QPixmap avatar = Settings::getInstance().getSavedAvatar(f->getToxId().toString());
+    ui->statusMessage->setText(f->getStatusMessage());
+    if(!avatar.isNull()) {
+        ui->avatar->setPixmap(avatar);
+    } else {
+        ui->avatar->setPixmap(QPixmap("://img/contact_dark.svg"));
```
Comment (TheNain38) [inline src/widget/about/aboutuser.cpp]: Normally it should be `:/img/contact_dark.svg` see #2543

Comment (agilob) [inline src/widget/about/aboutuser.cpp]: Fixed.

**Comment by ghost** [comment]
> I think `WHY ARE MY EYES BLEEDING` can be marked as done.

Yup, looks nice.

> > the list of aliases could become a scrollable list.
> 
> I don't get it.

Never mind, it was a bad idea.

Regarding the `Used aliases`, how long will the list be? In characters and/or number of aliases.

**Comment by agilob** [comment]
> Regarding the Used aliases, how long will the list be? In characters and/or number of aliases.

I don't really know how to implement this, what about a list comma separated? Or scrollable textedit max lines = 4 reverted chronological order?

**Comment by TheNain38** [comment]
@agilob The "WHY ARE MY EYES BLEEDING" can be marked done I think too, it looks nice.
A comma separated list isn't a good idea, because if the user sets comma in his name, then it will mess this up

**Comment by agilob** [comment]
screenshot5
screenshot6

**Comment by agilob** [comment]
OK, enough for today :)

screenshot8

**Comment by agilob** [comment]
> A comma separated list isn't a good idea, because if the user sets comma in his name, then is will mess this up

I was thinking about the same... I will try a 3line text area one line = one nick.

**Comment by TheNain38** [comment]
@agilob Thanks for making progress on this thing, would be really, really cool when finished :)

**Comment by ghost** [comment]
Or the aliases could be encapsulated in a box like the public key. Then the comma can't be confused with part of the nick and obscenely long aliases can be trimmed.

**Comment by ghost** [comment]
- The spacing between name and status is weird. I think that it should be displayed in a similar way as the main window (name  slightly bigger, status slightly smaller), vertically centred to the avatar.

screenshot from 2015-11-13 20-53-34
- Small bug: enable `Auto accept files`, dir chooser opens, click `Cancel` -> path in `Default directory` is left empty. It should be set to the `Default directory to save files` from qTox' config (or home).

**Comment by agilob** [comment]
screenshot10

Fixed both issues.

**Comment by ghost** [comment]
I'm getting random segfaults. Some times, the first time a password-protected profile is loaded with this branch (happened 4 times so far, but with the build from 2 commits ago)

> $ ./qtox 
> [21:24:27.650] src/persistence/settings.cpp:120 : Debug: "Loading settings from /home/bob/.config/tox/qtox.ini"
> [21:24:27.651] src/ipc.cpp:46 : Debug: Our global IPC ID is  xxx
> [21:24:27.651] src/ipc.cpp:64 : Debug: Attaching to the global shared memory
> [21:24:27.651] src/main.cpp:143 : Debug: built on:  20:43:02 Nov 13 2015 ( 1447443664 )
> [21:24:27.652] src/main.cpp:144 : Debug: commit:  09405f24934dd5e432812849974ddf0d4e18259f 
> 
> [21:24:27.652] src/ipc.cpp:125 : Debug: postEvent  "activate" to 0
> [21:24:28.637] src/nexus.cpp:82 : Debug: Starting up
> [21:24:37.652] src/persistence/profile.cpp:76 : Debug: Loading tox save  "/home/bob/.config/tox/testpass.tox"
> [21:24:37.802] src/core/coreencryption.cpp:83 : Warning: Decryption failed
> [21:24:37.802] src/persistence/profile.cpp:117 : Critical: Failed to decrypt the tox save file
> [21:24:43.577] src/persistence/profile.cpp:76 : Debug: Loading tox save  "/home/bob/.config/tox/testpass.tox"
> [21:24:43.807] src/persistence/settings.cpp:346 : Debug: "Saving global settings at /home/bob/.config/tox/qtox.ini"
> [21:24:43.858] src/persistence/settings.cpp:287 : Debug: Loading personnal settings from "/home/bob/.config/tox/testpass.ini"
> fish: “./qtox” terminated by signal SIGSEGV (Address boundary error)

or when I right-click a friend and select `Show details` (1 time)

> $ ./qtox 
> [21:38:31.174] src/persistence/settings.cpp:120 : Debug: "Loading settings from /home/bob/.config/tox/qtox.ini"
> [21:38:31.174] src/ipc.cpp:46 : Debug: Our global IPC ID is  xxx
> [21:38:31.175] src/ipc.cpp:64 : Debug: Attaching to the global shared memory
> [21:38:31.175] src/main.cpp:143 : Debug: built on:  21:30:45 Nov 14 2015 ( 1447533006 )
> [21:38:31.175] src/main.cpp:144 : Debug: commit:  734faa8a1f49bb77bc8e5c5d2a29b31afc42c1d5 
> 
> [21:38:31.175] src/ipc.cpp:125 : Debug: postEvent  "activate" to 0
> [21:38:32.634] src/nexus.cpp:82 : Debug: Starting up
> [21:38:34.929] src/persistence/profile.cpp:76 : Debug: Loading tox save  "/home/bob/.config/tox/fsaffdsa.tox"
> [21:38:34.930] src/persistence/settings.cpp:346 : Debug: "Saving global settings at /home/bob/.config/tox/qtox.ini"
> [21:38:34.981] src/persistence/settings.cpp:287 : Debug: Loading personnal settings from "/home/bob/.config/tox/fsaffdsa.ini"
> fish: “./qtox” terminated by signal SIGSEGV (Address boundary error)

**Comment by agilob** [comment]
Could you locally merge this branch into latest tux3/master? I think it's problem with trayicon not about window.

**Comment by ghost** [comment]
Will do.

Last thought for today: if you open the `Show details` window before a new friend is seen online, the window title and name are the public key. It doesn't get updated on the fly if the friend goes online, but it'd be nice if it did.

**Comment by PKEv** [comment]
Can save notes about the contact in the database?

**Comment by agilob** [comment]
Database schema needs to be rewritten. So, no, not now.

**Comment by ovalseven8** [comment]
Concerning the avatars:
Would be great if avatars will be about 160 x 160px. I think that's a nice size (for example Facebook is using it on profile pages). Of course, if the user has chosen a smaller avatar, the normal size of that avatar should be displayed (in the 160 x 160px box), so it isn't pixelated. 
Additionally, if you click on the avatar it should be possible to see the avatar in original size. Not sure about how to implement it. Perhaps something like 'Would you like to download the avatar?' or any other ideas?

**Comment by agilob** [comment]
Nah. Overkill.

**Comment by ovalseven8** [comment]
@agilob: What's overkill? Well, 130 x 130px is also good enough. But really, that should be possible.
People want to see the faces etc. And of course they should be able to see the full size avatar saved in the database what could be done with right-clicking on the avatar and click 'Save avatar'. What's wrong with that?

**Comment by ProMcTagonist** [comment]
Users want to see their friends' media, not our chrome. Current avatar in about window shown here isn't big enough.

**Comment by zetok** [comment]
> Current avatar in about window shown here isn't big enough.

It is big enough.

What you might want there, is displaying avatar in similar way to the way in which images are being displayed in chatform, perhaps with a slight adjustment to the logic, to display up to full image size, if size isn't bigger than half the size of screen.

**Comment by ovalseven8** [comment]
It's definitely not big enough. About 120 x 120px would be good.
And a way to watch the image in full size. Perhaps right-click and 'Save image' or how images are being displayed in chatform at the moment.

**Comment by agilob** [comment]
Sorry, I don't have much time now for everything. This PR can be considered 'done' and merged. Top message indicates what was implemented and what wasn't (gives reasons).

**Comment by tux3** [comment]
The about windows seems to work OK, but the style is going crazy for me. It's adding blue everywhere.

snap2

**Comment by agilob** [comment]
Either, remove parent from AbouUser constructor or redefine style to be applied on #name instead a class :<

@antis81 what was your solution you suggested once?

**Comment by tux3** [comment]
How come the blue style is being applied globally? It should be only for the contact list

**Comment by agilob** [comment]
I think I know how to fix it. w8

**Comment by agilob** [comment]
OK, fixed for me, could you test it now?

**Comment by tux3** [comment]
It works for me.
Except that it crashes on startup, but I think that's just because your branch is behind master, so it should be good to merge.

**Comment by antis81** [comment]
I wrote a static method `Widget::mainWidget()` some months ago, but dared about pushing it. I'll open new PR later today or tomorrow… :)

**Comment by agilob** [comment]
> Except that it crashes on startup, but I think that's just because your branch is behind master, so it should be good to merge.

It's far behind master, and behind IPC fix

Thanks @tux3  o/

**Pull request merged**